lib/http3/qpack.c

```diff
@@ -1300,8 +1300,9 @@ h2o_iovec_t h2o_qpack_flatten_request(h2o_qpack_encoder_t *_qpack, h2o_mem_pool_
 
     /* pseudo headers */
     flatten_known_header_with_static_lookup(&ctx, h2o_qpack_lookup_method, H2O_TOKEN_METHOD, method);
-    int is_connect = h2o_memis(method.base, method.len, H2O_STRLIT("CONNECT"));
-    if (!is_connect) {
+    /* CONNECT request that is not an RFC 9220 extended connect */
+    int old_style_connect = (h2o_memis(method.base, method.len, H2O_STRLIT("CONNECT")) && (protocol.base == NULL));
```

Comment:
```suggestion
    int old_style_connect = h2o_memis(method.base, method.len, H2O_STRLIT("CONNECT")) && protocol.base == NULL;
```
We do not add parens to mark precedence between = and && or && and ==.

Reply:
Done